pkg/sypgp/sypgp.go

```diff
@@ -349,6 +350,68 @@ func StorePubKey(e *openpgp.Entity) (err error) {
 	return
 }
 
+// compareLocalPubKey : compares a key ID with a string. return true if the
+// key / oldToken *dont* match.
+func compareLocalPubKey(e *openpgp.Entity, oldToken string) bool {
+	if fmt.Sprintf("%X", e.PrimaryKey.Fingerprint) != oldToken {
+		return true
+	}
+	return false
+}
+
+// RemovePupKey : will delete all keys with the ID of (toDelete string)
+func RemovePupKey(toDelete string) error {
+	//f, err := os.OpenFile(PublicPath(), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+	f, err := os.OpenFile(PublicPath(), os.O_APPEND, 0600)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	// read all the local public keys
+	elist, err := openpgp.ReadKeyRing(f)
+	if err != nil {
+		return fmt.Errorf("unable to read keyring: %v", err)
+	}
+
+	var newKeyList []openpgp.Entity
+
+	// sort throught them, and remove any that match toDelete
+	for i := range elist {
+		// if the elist[i] dose not match toDelete, then add it to newKeyList
+		if compareLocalPubKey(elist[i], toDelete) {
```

Comment:
this is what I meant before: only looking at this code, I would flag a bug, as it suggests that it's removing everything _but_ whatever matches `toDelete`.